storage/decorator.py

```diff
@@ -3,75 +3,63 @@
 # you may not use this file except in compliance with the Elastic License 2.0.
 
 import zlib
-from typing import Any, Callable, Iterator
+from typing import Any, Iterator
 
 from share import shared_logger
 
-from .storage import CommonStorageType
+from .storage import CommonStorageType, GetByLinesCallable
 
 
-def by_lines(
-    func: Callable[[CommonStorageType, int, Any, str, int], Iterator[tuple[bytes, int, int]]]
-) -> Callable[[CommonStorageType, int, Any, str, int], Iterator[tuple[bytes, int, int]]]:
+def by_lines(func: GetByLinesCallable[CommonStorageType]) -> GetByLinesCallable[CommonStorageType]:
```

Comment:
Nicer this way, yes :slightly_smiling_face: Is the `[CommonStorageType]` needed?

Reply:
because of `CommonStorage` being an abstract class: I have to use the related TypeVar/generic since the value passed to the decorators will be the concert implementation.
ie: passing the abstract class type is not enough to "catch" every concrete implementation